### PR TITLE
Describe v3.5 tags.

### PR DESCRIPTION
**Pull request by gluszczyk**
# Links

- [Jira](https://livechatinc.atlassian.net/browse/API-10298)

### CODE REVIEW

File: src/pages/management/configuration-api/v3.5/index.mdx [hunk 1]
```diff
@@ -2388,6 +2388,236 @@ curl -X POST \
...
+<CodeSample path={'REQUEST'}>
+
+```shell
+curl -X POST \
+  https://api.livechatinc.com/v3.5/configuration/action/create_tag \
+  -H 'Content-Type: application/json' \
+  -H 'Authorization: Bearer <your_access_token>' \
+  -d '{
+    "name": "docs_feedback",
+    "group_ids": [
```
Comment (ale-kacperczyk) [inline src/pages/management/configuration-api/v3.5/index.mdx:2458]: I don't think we have a rule as to how we present arrays, but to me, it doesn't look good when there are only 2 elements and each is in a new line. wdyt?

File: src/pages/management/configuration-api/v3.5/index.mdx [hunk 2]
```diff
@@ -2388,6 +2388,236 @@ curl -X POST \
...
+|                 |                                                                    |
+| --------------- | ------------------------------------------------------------------ |
+| Method URL      | `https://api.livechatinc.com/v3.5/configuration/action/delete_tag` |
+| Required scopes | `tags--all:rw, tags--groups:rw`                                    |
+
+#### Request
+
+| Parameter | Required | Data type | Notes                                                   |
+| --------- | -------- | --------- | ------------------------------------------------------- |
+| `name`    | yes      | `string`  | Matching the name of an existing tag is case-sensitive. |
```
Comment (ale-kacperczyk) [inline src/pages/management/configuration-api/v3.5/index.mdx:2490]:
```suggestion
| `name`    | yes      | `string`  | Matching the name of an existing tag is case sensitive. |
```

Comment (ale-kacperczyk) [reply src/pages/management/configuration-api/v3.5/index.mdx:2490]: This is correct, but it's more popular in BrE and we lean toward AmE

File: src/pages/management/configuration-api/v3.5/index.mdx [hunk 3]
```diff
@@ -2388,6 +2388,236 @@ curl -X POST \
...
+
+<CodeSample path={'REQUEST'}>
+
+```shell
+curl -X POST \
+  https://api.livechatinc.com/v3.5/configuration/action/delete_tag \
+  -H 'Content-Type: application/json' \
+  -H 'Authorization: Bearer <your_access_token>' \
+  -d '{
+    "name": "docs_feedback"
```
Comment (ale-kacperczyk) [inline src/pages/management/configuration-api/v3.5/index.mdx:2508]: I like the idea :D

File: src/pages/management/configuration-api/v3.5/index.mdx [hunk 4]
```diff
@@ -2388,6 +2388,236 @@ curl -X POST \
...
+        0,
+        42
+      ]
+    }
+  }'
+```
+
+</CodeSample>
+
+<CodeResponse type="configuration" version="v3.5" json="listTags"/>
```
Comment (ale-kacperczyk) [inline src/pages/management/configuration-api/v3.5/index.mdx:2563]: This is not rendered. Export it in `payloads/management/v3.5/configuration-api/index.js`

File: src/pages/data-reporting/reports-api/v3.5/index.mdx [hunk 5]
```diff
@@ -849,6 +849,70 @@ https://api.livechatinc.com/v3.5/reports/agents/performance \
...
 </Section>
 
+## Tags
+
+<Section>
+<Text>
+
+### Chat Usage
+
+Shows the total number of chats marked with each tag.
```
Comment (ale-kacperczyk) [inline src/pages/data-reporting/reports-api/v3.5/index.mdx:859]: Something's missing in this sentence. I guess we need a little more specificity as to what "each" refers to. E.g.:

> ...each tag within a license.

File: src/pages/management/webhooks/v3.5/index.mdx [hunk 6]
```diff
@@ -1906,6 +1906,136 @@ Informs that a group has been updated within a license. Only changed fields are
 
 </Section>
 
+<Section>
+
+<Text>
+
+### `tag_created`
+
+Informs that a tag has been created within a license. The payload contains the full state of the [tag](/management/configuration-api/v3.5/#tags) data structure, including empty fields.
```
[the remark this answers is not in the document]
Comment (gluszczyk) [reply src/pages/management/webhooks/v3.5/index.mdx:1915]: 👀 [hmmm](https://github.com/livechat/livechat-public-docs/blob/master/src/pages/management/webhooks/index.mdx?plain=1#L1809)

File: src/pages/management/configuration-api/v3.5/index.mdx (hunk 1 above)
Comment (gluszczyk) [reply src/pages/management/configuration-api/v3.5/index.mdx:2458]: but it's consistent with other displays of `group_ids` in this file like [here](https://github.com/livechat/livechat-public-docs/blob/master/src/pages/management/configuration-api/index.mdx?plain=1#L297)
let me know how to fix it 😉

File: src/pages/management/configuration-api/v3.5/index.mdx (hunk 2 above)
Comment (gluszczyk) [reply src/pages/management/configuration-api/v3.5/index.mdx:2490]: "whatever-keyboard-accepts" English is the best English :trollface:

File: src/pages/data-reporting/reports-api/v3.5/index.mdx (hunk 5 above)
Comment (gluszczyk) [reply src/pages/data-reporting/reports-api/v3.5/index.mdx:859]: more like "each tag listed in the report", wdyt? cause in the request you can specify groups to list tags from

Comment (ale-kacperczyk) [reply src/pages/data-reporting/reports-api/v3.5/index.mdx:859]: Sounds ok 👍🏻

File: src/pages/management/configuration-api/v3.5/index.mdx (hunk 1 above)
Comment (ale-kacperczyk) [reply src/pages/management/configuration-api/v3.5/index.mdx:2460]: Here's a counter-example: https://developers.livechat.com/docs/management/configuration-api#update-auto-access

All in all, I think that in most cases it's ugly formatted (each id in a separate line). We can keep your request formatted in the same way for better consistency, but I would rethink formatting some time in the future.

File: src/pages/management/webhooks/v3.5/index.mdx (hunk 6 above)
[the remark this answers is not in the document]
Comment (ale-kacperczyk) [reply src/pages/management/webhooks/v3.5/index.mdx:1915]: Yeah, I know, but I think there are more examples with "was" (I remember changing it to Present Simple). I need to update those with Pr. Perfect to match the rest

File: src/pages/management/configuration-api/v3.5/index.mdx [hunk 7]
```diff
@@ -2388,6 +2388,236 @@ curl -X POST \
...
+| --------------- | ----------------------------------------------------------------- |
+| Method URL      | `https://api.livechatinc.com/v3.5/configuration/action/list_tags` |
+| Required scopes | `tags--all:ro, tags--groups:ro`                                   |
+
+#### Request
+
+| Parameter           | Required | Data type | Notes                                                                  |
+| ------------------- | -------- | --------- | ---------------------------------------------------------------------- |
+| `filters`           | No       | `object`  | Possible request filters.                                              |
+| `filters.group_ids` | Yes      | `int[]`   | IDs of the groups from which you want to list Tags.                    |
```
[the remark this answers is not in the document]
Comment (ale-kacperczyk) [reply src/pages/management/configuration-api/v3.5/index.mdx:2540]: Yeah, we used capitalize "Agents", but the content rules in the product have changed, yet there are still a lot of places with Agent

File: src/pages/data-reporting/changelog/index.mdx [hunk 8]
```diff
@@ -23,11 +23,15 @@ The developer preview version provides a preview of the upcoming changes to the
...
 
-- There are new filters in all reports: 
+- There are new filters in all reports:
   - `filters.customer_countries.values`
   - `filters.customer_countries.exclude_values`
   - `filters.customer_countries.require_every_value`
 
+### Tags
+
+- There's a new method, [**Tags Chat Usage**](/data-reporting/reports-api/v3.5/#chat_usage), which shows how many chats are marked with each listed tag.
```
Comment (ale-kacperczyk) [inline src/pages/data-reporting/changelog/index.mdx:33]:
```suggestion
- There's a new method, [**Tags Chat Usage**](/data-reporting/reports-api/v3.5/#chat-usage), which shows how many chats are marked with each listed tag.
```

File: src/pages/data-reporting/reports-api/v3.5/index.mdx [hunk 9]
```diff
@@ -56,6 +56,7 @@ You can authorize your calls to the Reports API using one of the following metho
...
 | **Chats**  | [`duration`](#duration) [`tags`](#tags) [`total_chats`](#total-chats) [`ratings`](#ratings) [`ranking`](#ranking) [`engagement`](#engagement) [`greetings_conversion`](#greetings-conversion) [`surveys`](#surveys) [`response_time`](#response-time) [`first_response_time`](#first-response-time) |
 | **Agents** | [`availability`](#availability) [`performance`](#performance)                                                                                                                                                                                                                                       |
+| **Tags**   | [`chat_usage`](#chat_usage)                                                                                                                                                                                                                                                                         |
```
Comment (ale-kacperczyk) [inline src/pages/data-reporting/reports-api/v3.5/index.mdx:59]:
```suggestion
| **Tags**   | [`chat_usage`](#chat-usage)                                                                                                                                                                                                                                                                         |
```

File: src/pages/messaging/agent-chat-api/v3.5/rtm-pushes/index.mdx [hunk 10]
```diff
@@ -1032,6 +1032,52 @@ Informs that a group was updated within a license. The push payload only include
 
 </CodeResponse>
 
+### `tag_created`
+
+Informs that a tag has been created within a license. The payload contains the full state of the [tag](/management/configuration-api/v3.5/#tags) data structure, including empty fields.
```
Comment (ale-kacperczyk) [inline src/pages/messaging/agent-chat-api/v3.5/rtm-pushes/index.mdx:1037]:
```suggestion
Informs that a tag was created within a license. The payload contains the full state of the [tag](/management/configuration-api/v3.5/#tags) data structure, including empty fields.
```

File: src/pages/messaging/agent-chat-api/v3.5/rtm-pushes/index.mdx [hunk 11]
```diff
@@ -1032,6 +1032,52 @@ Informs that a group was updated within a license. The push payload only include
...
+{
+  "name": "docs_feedback"
+}
+```
+
+</CodeResponse>
+
+### `tag_updated`
+
+Informs that a tag has been updated within a license.
```
Comment (ale-kacperczyk) [inline src/pages/messaging/agent-chat-api/v3.5/rtm-pushes/index.mdx:1068]:
```suggestion
Informs that a tag was updated within a license.
```

File: src/pages/messaging/agent-chat-api/v3.5/rtm-pushes/index.mdx [hunk 12]
```diff
@@ -1032,6 +1032,52 @@ Informs that a group was updated within a license. The push payload only include
...
+  "created_at": "2019-12-09T12:01:18.909000Z",
+  "author_id": "smith@example.com"
+}
+```
+
+</CodeResponse>
+
+### `tag_deleted`
+
+Informs that a tag has been deleted within a license.
```
Comment (ale-kacperczyk) [inline src/pages/messaging/agent-chat-api/v3.5/rtm-pushes/index.mdx:1054]:
```suggestion
Informs that a tag was deleted within a license.
```

File: src/pages/management/webhooks/v3.5/index.mdx [hunk 13]
```diff
@@ -1906,6 +1906,136 @@ Informs that a group has been updated within a license. Only changed fields are
...
+
+</Section>
+
+<Section>
+
+<Text>
+
+### `tag_updated`
+
+Informs that a tag has been updated within a license.
```
Comment (ale-kacperczyk) [inline src/pages/management/webhooks/v3.5/index.mdx:2002]:
```suggestion
Informs that a tag was updated within a license.
```

Comment (ale-kacperczyk) [reply src/pages/management/webhooks/v3.5/index.mdx:2002]: and so on

**Review by ale-kacperczyk — APPROVED** [review]

**Review by janowsiany — APPROVED** [review]

**Pull request merged**